hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/defaultkycstatus/DefaultKycStatusCall.java

```diff
@@ -50,10 +50,8 @@ public DefaultKycStatusCall(
     @Override
     protected @NonNull PricedResult resultOfViewingToken(@Nullable final Token token) {
         requireNonNull(token);
-        return gasOnly(
-                fullResultsFor(SUCCESS, gasCalculator.viewGasRequirement(), token.accountsKycGrantedByDefault()),
-                SUCCESS,
-                true);
+        final boolean kycStatus = !token.hasKycKey() || token.accountsKycGrantedByDefault();
```

Comment:
Perhaps I misunderstood but where is this logic?
> If the related token is missing the kycKey the status is KycNotApplicable which is represented as a boolean true. We return false when the token has kycKey where the status is Revoked

Reply:
The return value for this function is based on the presence of the kycKey. Having the key means the default kyc status is Revoked (false)  and missing it means that kyc is not applicable (true) thus the negative `!token.hasKycKey()`.
The expression takes into consideration the value of the `defaultKycStatus` (that is always false) to ensure that if at any moment we decide to change the services behaviour for this field this function will work as expected.